### PR TITLE
Add support for multiple (External)Capturable types.

### PR DESCRIPTION
**Pull request by pchote**
Also replaces the ally/neutral/enemy flags with a Stance enum.

### CODE REVIEW

**Review by penev92 — APPROVED** [review]
👍

File: OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs [hunk 1]
```diff
@@ -1003,6 +1004,55 @@ internal static void UpgradeActorRules(ModData modData, int engineVersion, ref L
...
+								stance &= ~Stance.Ally;
+
+							node.Value.Nodes.Remove(allowAllies);
+						}
+
+						var allowNeutral = node.Value.Nodes.FirstOrDefault(n => n.Key == "AllowNeutral");
+						if (allowNeutral != null)
+						{
+							if (FieldLoader.GetValue<bool>("AllowNeutral", allowNeutral.Value.Value))
+								stance |= Stance.Neutral;
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs]: Does this even make sense, as you initialized `stance` with `Stance.Neutral | Stance.Enemy` (same for "AllowEnemies")?

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
Looks good to me, but I'd like to see @abcdefg30's question at least answered before we merge this.

**Comment by reaperrr** [comment]
> [15:43] <+pchote> reaperrr: it doesn't matter either way
[15:44] <+pchote> the behaviour in both cases is the same, its just a tradeoff of having easier to read but slightly redundant code vs more efficient but less obvious

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/d2b87a0c439abf9e0dcfcb6b5bba4200f8586de9...6a6dd22860b8199286df613f07f5f44e73d8b477

**Pull request merged**